PKHeX.Core/Legality/Encounters/EncounterGenerator.cs

```diff
@@ -129,7 +129,7 @@ private static IEnumerable<IEncounterable> GenerateFilteredEncountersPIDIV(PKM p
         private static IEnumerable<IEncounterable> GenerateRawEncounters(PKM pkm)
         {
             int ctr = 0;
-            if (pkm.WasLink)
+            if (pkm.WasLink && pkm.GenNumber == 6)
             {
                 foreach (var z in getValidLinkGifts(pkm))
```

Comment:
I recommend simply handling the null value returned by `getValidLinkGifts(pkm)`. Checking to see if the number is 6 is something the function does. If in the future the function was updated to support another generation, this would have to be changed as well, whereas nothing would have to change if you handled the null instead.